src/pages/HomePage.tsx

```diff
@@ -1,9 +1,28 @@
-import React from 'react';
+import React, { useState, useEffect } from "react";
+import LoadingOverlay from "../components/LoadingOverlay";
 
 export default function HomePage() {
+  const [isLoadingVisible, setIsLoadingVisible] = useState(false);
+
+  useEffect(() => {
+    // Check if it's the user's first visit for the session using sessionStorage
+    const isFirstVisit = sessionStorage.getItem("isFirstVisit");
+
+    if (!isFirstVisit) {
+      setIsLoadingVisible(true);
+      sessionStorage.setItem("isFirstVisit", "no");
+      setTimeout(() => {
+        setIsLoadingVisible(false);
+      }, 3200);
+    }
```

Comment:
Seems like we are using SessionStorage here to check whether is it the user's first time visiting the website and then subsequently rendering `LoadingOverlay` component to introduce framer motion effect if it is their first time?
I'm not familar with SessionStorage, seems like sessionStorage is cleared whenever a user navigates away from the page. So that's good and is what we're looking for here right?
@LAMEfreak

Reply:
correct, otherwise the overlay will trigger every time you navigate from any other page in the web app to the homepage. using session storage tracks it by browser tab, as long as it's the same tab and you've seen the overlay effect once, you won't see it again for subsequent visits to the homepage. each tab is a separate 'session'